src/Serilog.Extensions.Hosting/SerilogHostBuilderExtensions.cs

```diff
@@ -26,16 +26,27 @@ namespace Serilog
     /// </summary>
     public static class SerilogHostBuilderExtensions
     {
+        // Used internally to pass information through the container.
+        class RegisteredLogger
```

Comment:
I don’t understand why this hack is needed. Could you explain please?

Reply:
@sungam3r, 
`ILogger` needs to be registered as singleton. The registration method on MEDI is factory (to be able to implement the title feature). Registering singleton with factory method is subject to [automatic disposal](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-3.1#service-registration-methods) of the instantiated object that is undesirable since the disposing of `ILogger` handled by `SerilogLoggerFactory` (passing `dispose=true` to it constructor).
AFAIC, all this ceremony needed to correctly handle `preserverStaticLogger=false` otherwise the disposal could be handled by the DI container itself..

Reply:
Quite hard to understand to one who has no some background knowledge about all this stuff. I guessed that the library should consider all combinations of use. Maybe we should add your comment to the code? It’s hard to understand what’s going on without it.

Reply:
Added a comment paraphrasing Sergey's - thanks for the suggestion 👍